nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -899,12 +903,12 @@ impl<'a> BlueprintBuilder<'a> {
         let mut num_zones_expunged = 0;
 
         let mut disks_to_expunge = Vec::new();
-        for disk in editor.disks(DiskFilter::All) {
+        for disk in editor.disks(BlueprintPhysicalDiskDisposition::any) {
```

Comment:
Could this be 

```suggestion
        for disk in editor.disks(BlueprintPhysicalDiskDisposition::is_in_service) {
```

instead of needing a `match` within the loop?

Reply:
Derp. Done in 55a65f8d01ed8e54a7ade5b84eeae83a937956dd. Note that I had to keep the clone into a Vec because of the mutable borrow on `editor` inside the loop. Same for another change in the same commit.